apps/sr/tid1500reader.cxx

```diff
@@ -0,0 +1,218 @@
+        }
+        st.gotoParent();
+        st.gotoChild();
+
+        Json::Value measurementItems(Json::arrayValue);
+        while (st.gotoNext()){
+          if (st.getCurrentContentItem().getNumericValuePtr() != NULL) {
+            DSRNumericMeasurementValue measurementValue = st.getCurrentContentItem().getNumericValue();
+
+            Json::Value localMeasurement;
+            localMeasurement["value"] = measurementValue.getNumericValue().c_str();
+
+            localMeasurement["units"]["CodeValue"] = measurementValue.getMeasurementUnit().getCodeValue().c_str();
+            localMeasurement["units"]["CodeMeaning"] = measurementValue.getMeasurementUnit().getCodeMeaning().c_str();
+            localMeasurement["units"]["CodingSchemeDesignator"] = measurementValue.getMeasurementUnit().getCodingSchemeDesignator().c_str();
+
+            localMeasurement["quantity"]["CodeValue"] = st.getCurrentContentItem().getConceptName().getCodeValue().c_str();
+            localMeasurement["quantity"]["CodeMeaning"] = st.getCurrentContentItem().getConceptName().getCodeMeaning().c_str();
+            localMeasurement["quantity"]["CodingSchemeDesignator"] = st.getCurrentContentItem().getConceptName().getCodingSchemeDesignator().c_str();
+
+            if(st.gotoNamedChildNode(CODE_DCM_Derivation)){
+              localMeasurement["derivationModifier"]["CodeValue"] = st.getCurrentContentItem().getCodeValue().getCodeValue().c_str();
+              localMeasurement["derivationModifier"]["CodeMeaning"] = st.getCurrentContentItem().getCodeValue().getCodeMeaning().c_str();
+              localMeasurement["derivationModifier"]["CodingSchemeDesignator"] = st.getCurrentContentItem().getCodeValue().getCodingSchemeDesignator().c_str();
+              st.gotoParent();
+            }
+            measurementItems.append(localMeasurement);
+          }
+        }
+        measurement["measurementItems"] = measurementItems;
+        measurements.append(measurement);
+      }
+    } while (st.gotoNextNamedNode(CODE_DCM_MeasurementGroup));
+  }
+  return measurements;
+}
+
+int main(int argc, char** argv){
+  PARSE_ARGS;
+
+  Json::Value metaRoot;
+
+  DcmFileFormat sliceFF;
+  CHECK_COND(sliceFF.loadFile(inputSRFileName.c_str()));
+  DcmDataset* dataset = sliceFF.getDataset();
+
+  TID1500_MeasurementReport report(CMR_CID7021::ImagingMeasurementReport);
+  DSRDocument doc;
+
+  CHECK_COND(doc.read(*dataset));
+
+
+  OFString temp;
+  doc.getSeriesDescription(temp);
+  metaRoot["SeriesDescription"] = temp.c_str();
+  doc.getSeriesNumber(temp);
+  metaRoot["SeriesNumber"] = temp.c_str();
+  doc.getInstanceNumber(temp);
+  metaRoot["InstanceNumber"] = temp.c_str();
+
+  cout << "Number of verifying observers: " << doc.getNumberOfVerifyingObservers() << endl;
+
+  OFString observerName, observingDateTime, organizationName;
+  if (doc.getNumberOfVerifyingObservers() != 0) {
```

Comment:
I did not realize this, but now that I looked - it seems that there are two places where observer is recorded. The one I used for writing TID1500 is TID1001 http://dicom.nema.org/medical/dicom/current/output/chtml/part16/chapter_A.html#sect_TID_1001 Observation context: https://github.com/QIICR/dcmqi/blob/master/apps/sr/tid1500writer.cxx#L78-L84

I agree that in the future we should synchronize this with `DCM_Verifying/Verification*` items, but for now I think it will be better if reader is symmetric with the writer.